### DISCUSSION TITLE
server.resetHandlers before each test

### DISCUSSION THREAD

**Discussion by gvocale** [mswjs/msw#891] [Q&A]
I am mocking GraphQL requests made server side with Next.js `getServerSideProps` hook via `setupServe(...handlers)` following the example provided by Next.js https://github.com/vercel/next.js/tree/canary/examples/with-msw.

I would like to reset a specific resolver before each Cypress test to test different data scenarios. Is there a way to trigger `server.resetHandlers` from a `before` hook into a Cypress test?

I have found this stack overflow answer https://stackoverflow.com/questions/47631821/mocking-server-for-ssr-react-app-e2e-tests-with-cypress-io using MockTTP.

Is it possible to do the same with msw?

**Comment by kettanaito** [comment]
You can call `window.worker.resetHandlers()` as long as you propagate the created worker on the `window` object.

```js
// your-app.js
const worker = setupWorker(...handlers)
window.worker = worker
```

```js
// your.test.js
before(async () => {
  await cy.window().then((win) => {
    // Resetting handlers is a synchronous action.
    win.worker.resetHandlers()
  })
})
```

The usage is the same as with the [runtime request handlers with Cypress](https://mswjs.io/docs/api/setup-worker/use#runtime-request-handler).
